net/sslh/src/opnsense/mvc/app/controllers/OPNsense/Sslh/forms/settings.xml

```diff
@@ -0,0 +1,197 @@
+<form>
+    <field>
+        <id>settings.enabled</id>
+        <label>Enable SSLH</label>
+        <type>checkbox</type>
+        <help><![CDATA[
+            This will enable the SSLH service.
+            ]]>
+        </help>
+    </field>
+    <field>
+        <id>settings.listen_addresses</id>
+        <label>Listen Addresses</label>
+        <help><![CDATA[
+            <b>-p, --listen</b>  address:port<br>
+            Interface and port on which to listen, e.g. foobar:443, where
+            foobar is the name of an interface (typically the IP address on
+            which the Internet connection ends up).<br>
+            <br>
+            This can be specified several times to bind sslh to several
+            addresses.
+            ]]>
+        </help>
+        <style>tokenize</style>
+        <allownew>true</allownew>
+        <hint>hostname:443,hostname:1194</hint>
```

Comment:
does the tokenizer support hints in this way that are easily distinguishable for the user? Usually we provide a default that works like localhost:443 and since the field can't/shouldn't be empty that is all that is required to get started

Reply:
It does, looks like this:
![image](https://user-images.githubusercontent.com/7823088/148314020-722e3ee6-1c23-4223-beed-ca7f71456207.png)

I chose the hint method since the documentation explains thusly:
```
Interface and port on which to listen, e.g. foobar:443, where
foobar is the name of an interface (typically the IP address on
which the Internet connection ends up).
```

Which would be complicated to accomplish via the model defined default. In my cursory test using `localhost:443` doesn't work.

I'm not married to the hint if a proper default could be defined in the model.

Reply:
The hint is fine, but I would suggest avoiding the second entry altogether to not confuse users how tokenizer input should look like. It will lead to questions later.

localhost:443 probably will not work because the web GUI listens on it.

But the real question here is how the model is designed: do you need this entry in the configuration? If not what is the default or if you require it from the model you will force an initial model migration to fail if you don't set a default value which masks the hint you put there to begin with. :) hints are mostly useful for optional fields or when you properly embed all the constraints in the model (if enabled the hostname cannot be empty).

Reply:
The combination of required field without a default and how it would affect model migration was not something I was aware of. I've set a sane/functional default in the model, and removed the hint.